src/App.vue

```diff
@@ -86,17 +117,50 @@ export default {
       if (this.stopPoint != null  || this.lineId != undefined) {
         getPlatforms(this.stopPoint)
           .then(response => {
-            this.timetable = response;
+            this.timetable = response
             
             const lineTimetable = this.timetable.filter(x => {
               return x.lineId === this.lineId
             });
 
-            this.platforms = [...new Set(lineTimetable.map(i => i.platformName))]
-              .sort((a, b) => (a[a.length - 1] > b[b.length - 1]) ? 1 : -1)
-          })
-          .catch(error => alert(error.name))
-      }      
+            const duplicatedPlatforms = lineTimetable.map(i => {
+              return { name: i.platformName, value: i.platformName }
+            })
+
+            function dedupe(arr) {
+              return arr.reduce(function(p, c) {
+                // create an identifying id from the object values
+                var id = [c.name, c.value].join('|')
+                // if the id is not found in the temp array
+                // add the object to the output array
+                // and add the key to the temp array
+                if (p.temp.indexOf(id) === -1) {
+                  p.out.push(c);
+                  p.temp.push(id);
+                }
+                return p;
+                //return the deduped array
+              }, {
+                temp: [],
+                out: []
+              }).out;
+            }
```

Comment:
Was there a reason the map method wasn't working?

Happy for it to be done this but if we should be declaring this function in a utilities file and then importing it rather than declaring it inside another function call.